tests/scenarios/test_generic_nimbus.py

```diff
@@ -20,19 +24,33 @@ def check_branch_in_telemetry(telemetry_event_check, experiment_json, request, e
 @then("The Experiment is unenrolled via the about:studies page")
 def unenroll_via_studies_page(selenium, experiment_json):
     selenium.get("about:studies")
-    WebDriverWait(selenium, 60).until(
-        EC.visibility_of_element_located((By.CSS_SELECTOR, ".study-name")),
-        message="Experiment not shown on about:studies.",
-    )
-    items = selenium.find_elements(By.CSS_SELECTOR, ".study-name")
-    for item in items:
-        if experiment_json["userFacingName"] in item.text:
-            selenium.find_element(By.CSS_SELECTOR, ".remove-button").click()
+    timeout = timeout = time.time() + 60
+    while time.time() < timeout:
+        try:
+            items = selenium.find_elements(By.CSS_SELECTOR, ".study-name")
+            for item in items:
+                if experiment_json["userFacingName"] in item.text:
+                    selenium.find_element(By.CSS_SELECTOR, ".remove-button").click()
+        except NoSuchElementException:
+            time.sleep(2)
+            selenium.refresh()
+            pass
+        else:
+            logging.info("Experiment unenrolled")
+            break
```

Comment:
Same thing here, is it possible to control program flow without using exceptions?

Reply:
Hmm, let me try. This is an old habit I need to break, I was trying with `next()`.